doc/modules/neighbors.rst

```diff
@@ -724,3 +724,211 @@ added space complexity in the operation.
 
     .. [2] `Wikipedia entry on Neighborhood Components Analysis
       <https://en.wikipedia.org/wiki/Neighbourhood_components_analysis>`_
+
+
+.. _lmnn:
+
+Large Margin Nearest Neighbor
+=============================
+
+.. sectionauthor:: John Chiotellis <johnyc.code@gmail.com>
+
+Large Margin Nearest Neighbor (LMNN, :class:`LargeMarginNearestNeighbor`) is
+a metric learning algorithm which aims to improve the accuracy of
+nearest neighbors classification compared to the standard Euclidean distance.
+
+.. |lmnn_illustration_1| image:: ../auto_examples/neighbors/images/sphx_glr_plot_lmnn_illustration_001.png
+   :target: ../auto_examples/neighbors/plot_lmnn_illustration.html
+   :scale: 50
+
+.. |lmnn_illustration_2| image:: ../auto_examples/neighbors/images/sphx_glr_plot_lmnn_illustration_002.png
+   :target: ../auto_examples/neighbors/plot_lmnn_illustration.html
+   :scale: 50
+
+.. centered:: |lmnn_illustration_1| |lmnn_illustration_2|
+
+
+For each training sample, the algorithm fixes :math:`k` "target neighbors",
+namely the :math:`k`-nearest training samples (as measured by the Euclidean
+distance) that share the same label. Given these target neighbors, LMNN
+learns a linear transformation of the data by optimizing a trade-off between
+two goals. The first one is to make each (transformed) point closer to its
+target neighbors than to any differently-labeled point by a large margin,
+thereby enclosing the target neighbors in a sphere around the reference
+sample. Data samples from different classes that violate this margin are
+called "impostors". The second goal is to minimize the distances of each
+sample to its target neighbors, which can be seen as some sort of
+regularization.
+
+Classification
+--------------
+
+Combined with a nearest neighbors classifier (:class:`KNeighborsClassifier`),
+this method is attractive for classification because it can naturally
+handle multi-class problems without any increase in the model size, and only
```

Comment:
Not sure this statement here is fully bullet-proof. As a memory-based algorithm, kNN-type procedure use the training data as 'model', which is why the notion of 'model size' may be less immediately clear in this context.

Reply:
That's true, but even in that case, it holds that nothing needs to be modified in the algorithm to handle multi-class problems (in contrast to e.g. SVMs). Should we change maybe `any increase in the model size` to `any change in the algorithm`. I thing this paragraph was by @bellet ?

Reply:
Well, the idea is that the model size of LMNN (as well as kNN) does not depend on the number of classes. In the sense that you can have as many labels as you want in the training data, the number of parameters to learn in LMNN remains the same (and the model size/complexity of kNN remains the same for fixed training set size).